index.js

```diff
@@ -130,6 +130,24 @@ function notLookingFor(msg, text, chatError, notFoundError) {
   }
 }
 
+function giveHelp(msg) {
+  answer = ""
+  for (command in actions) {
```

Comment:
`continue` should only be used to simplify your code, not to make it longer. Also, line 141 cannot throw any exceptions anyway, so those `try` and `catch` clauses have no use. Have you considered  replacing lines 135-147 with
``` javascript
for (command in actions)
  if (actions[command].description !== undefined)
    answer += `/${command}: ${actions[command].description}\n`;
```

Reply:
I don't agree @FoxySeta. 
The syntax you propose is quite bug-prone and should never be used except when you do competitive programming.

So the tradeoff here is writing 2 less lines and gain 30 minutes or hours of debugging in case of bugs.
I prefer time. 

And the try catch is useful if the json file is malformed, or someone changes the actions global later, you could discover the error in no time.

Reply:
I could opt for a more general (and thus safer) check:
```javascript
for (command in actions)
  if (actions[command] && actions[command].description)
    answer += `/${command}: ${actions[command].description}\n`;
```
But beside that, I still do not consider this shorter version error-prone at all. Also, there are no more exceptions to be caught by a `catch` clause.

Reply:
Ok, I'll change as requested tomorrow

Reply:
But I'd say it is, don't know how to convince you, maybe you should ask to a senior engineer and then you would believe me.

Reply:
@Flecart the code you put forward is flawed. Just set any field in `actions` to a nullish value and it'll break. That's because it reads the description property without checking for nullishness. Furthermore try-catch block is useless as the only possible error (an invalid property access) would have already been thrown a few lines above in the if guard where you access the same property.

Even from a structural standpoint any data error should not be handled by the function rather by its caller. It's a job of the procedure which reads the json file to check for input data correctness. Assuming input data is correct removes needless complexity, redundant checks (as the `actions` data is never modified after load) and lots of duplicated code which would eventually pop up in each procedure accessing these properties.

Nonetheless @FoxySeta version definitely is safe against any corrupted input (both a field being nullish and a property missing) at the "expese" of one line of code and two memory reads. If the choice is between the maintainer's code and yours, just do as you're told and politely shut the fuck up.

Reply:
@lucat1 yeah what you say about the try catch is right, i agree now, but i don't think you are right with  the style for the the for cycle without braces and [if statements without braces](https://stackoverflow.com/questions/2125066/is-it-a-bad-practice-to-use-an-if-statement-without-curly-braces), that has caused a [major security bug](https://www.imperialviolet.org/2014/02/22/applebug.html), so indeed it's just a bad practice.
I don't think you are right about the [callback hell](http://callbackhell.com/) at notLookingfor function.

And what about your tendence not to [modularize your code](http://singlepageappbook.com/maintainability1.html#:~:text=What%20is%20modular%20code%3F,not%20just%20about%20code%20organization.)?

I don't think you can politely tell me to shut the fuck up. You're not polite, and you are just rejecting some costructive criticism for some subjective choices. at the end the choice is yours but i hope the reality of the facts are clear and the resources can give you a different view on this topic.

And i'd say i'm deeply offended by your answer and don't think i will contribute here ever again.